pkg/cmd/task/describe.go

```diff
@@ -0,0 +1,174 @@
+// Copyright © 2019 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package task
+
+import (
+	"fmt"
+	"text/tabwriter"
+	"text/template"
+
+	"github.com/jonboulle/clockwork"
+	"github.com/spf13/cobra"
+	"github.com/tektoncd/cli/pkg/cli"
+	"github.com/tektoncd/cli/pkg/formatted"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	cliopts "k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+const describeTemplate = `Name:	{{ .Task.Name }}
+Namespace:	{{ .Task.Namespace }}
+
+Input Resources
+{{- if not .Task.Spec.Inputs }}
+No input resources
+{{- else }}
+{{- if eq (len .Task.Spec.Inputs.Resources) 0 }}
+No input resources
+{{- else }}
+NAME	TYPE
+{{- range $ir := .Task.Spec.Inputs.Resources }}
+{{ $ir.Name }}	{{ $ir.Type }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+Output Resources
+{{- if not .Task.Spec.Outputs }}
+No output resources
+{{- else }}
+{{- if eq (len .Task.Spec.Outputs.Resources) 0 }}
+No output resources
+{{- else }}
+NAME	TYPE
+{{- range $or := .Task.Spec.Outputs.Resources }}
+{{ $or.Name }}	{{ $or.Type }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+Params
+{{- if not .Task.Spec.Inputs }}
+No params
+{{- else }}
+{{- if eq (len .Task.Spec.Inputs.Params) 0 }}
+No params
+{{- else }}
+NAME	TYPE
+{{- range $p := .Task.Spec.Inputs.Params }}
+{{ $p.Name }}	{{ $p.Type }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+Steps
+{{- if eq (len .Task.Spec.Steps) 0 }}
+No steps
+{{- else }}
+NAME
+{{- range $step := .Task.Spec.Steps }}
+{{ $step.Name }}
+{{- end }}
+{{- end }}
+
+Runs
```

Comment:
Think it would be better to be more explicit by using `TaskRuns` instead of `Runs`.